File: server/server/Models/Ticket.cs
﻿namespace server.Models
{
    public enum TicketStatus
    {
        Pending,
        Paid,
        Win
    }
    public class Ticket
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }

        public int GiftId { get; set; }
        public Gift Gift { get; set; }

        public DateOnly OrderDate { get; set; }
        public DateOnly? PayDate { get; set; }
        public TicketStatus Status { get; set; } = TicketStatus.Pending;

    }
}

File: client/src/app/domain/ticket.ts
import { Product } from "./product";
import { User } from "./user";

enum TicketStatus {
    Pending,
    Paid,
    Win
}

export interface Ticket {
    id: number;
    user: User;
    gift: Product;
    orderDate: Date;
    status: TicketStatus;
}